src/webapp/pages/settings/useNotifications.ts

```diff
@@ -0,0 +1,104 @@
+    const [notifications, setNotifications] = useState<NotificationAttrs[]>([]);
+    const [isLoading, setIsLoading] = useState(true);
+
+    const fetchNotifications = useCallback(async () => {
+        setIsLoading(true);
+        try {
+            const notifications = await compositionRoot.notifications.get().toPromise();
+            setNotifications(notifications);
+        } finally {
+            setIsLoading(false);
+        }
+    }, [compositionRoot]);
+
+    const saveNotifications = useCallback(
+        async (notifications: NotificationAttrs[]) => {
+            setIsLoading(true);
+            try {
+                await compositionRoot.notifications.save(toNotification(notifications)).toPromise();
+                setNotifDetailsDialog(undefined);
+            } catch (err: any) {
+                snackbar.error((err && err.message) || err.toString());
+            } finally {
+                await fetchNotifications();
+            }
+        },
+        [compositionRoot.notifications, fetchNotifications, snackbar]
+    );
+
+    const editNotification = useCallback(
+        (notifId: string) => {
+            const notification = notifications.find(({ id }) => notifId === id);
+            if (!notification) return;
+
+            setNotifDetailsDialog({
+                initialNotification: { ...notification },
+                onClose: () => setNotifDetailsDialog(undefined),
+                onSave: notification => saveNotifications([notification]),
+            });
+        },
+        [notifications, saveNotifications]
+    );
+
+    const newNotification = useCallback(() => {
+        setNotifDetailsDialog({
+            onClose: () => setNotifDetailsDialog(undefined),
+            onSave: notification => saveNotifications([notification]),
+        });
+    }, [saveNotifications]);
+
+    const deleteNotifications = useCallback(
+        async (notifIds: string[]) => {
+            setIsLoading(true);
+            const notificationsToDelete = notifications.filter(({ id }) => notifIds.includes(id));
+            try {
+                await compositionRoot.notifications.delete(toNotification(notificationsToDelete)).toPromise();
+                snackbar.success(i18n.t("Successfully deleted notifications"));
+            } catch (err: any) {
+                snackbar.error((err && err.message) || err.toString());
+            } finally {
+                await fetchNotifications();
+            }
+        },
+        [notifications, compositionRoot.notifications, snackbar, fetchNotifications]
+    );
+
+    useEffect(() => {
+        fetchNotifications();
+    }, [fetchNotifications]);
+
+    useEffect(() => {
+        setNotifDetailsDialog(props =>
+            props
+                ? {
+                      ...props,
+                      isLoading: isLoading,
+                  }
+                : undefined
+        );
+    }, [isLoading]);
```

Comment:
useEffect blocks are anonymous. We have two approaches, either move it to a function or when the function is simple and that's considered overkill, add a comment explaining what it does.